integration/query_array_test.go

```diff
@@ -137,3 +140,110 @@ func TestQueryArraySize(t *testing.T) {
 		})
 	}
 }
+
+func TestQueryArrayDotNotation(t *testing.T) {
+	t.Parallel()
+	providers := []shareddata.Provider{shareddata.Composites}
+	ctx, collection := setup(t, providers...)
+
+	_, err := collection.InsertMany(ctx, []any{
+		bson.D{{"_id", "array-double"}, {"value", bson.A{float64(1)}}},
+		bson.D{{"_id", "array-strings"}, {"value", bson.A{"foo", "bar", "baz"}}},
+		bson.D{
+			{"_id", "document-array-field"},
+			{"value", bson.D{{"array", bson.A{int32(0), nil}}}},
+		},
+		bson.D{
+			{"_id", "document-document-field"},
+			{"value", bson.D{{"document", bson.D{{"foo", nil}}}}},
+		},
+	})
+	require.NoError(t, err)
+
+	for name, tc := range map[string]struct {
+		filter      bson.D
+		expectedIDs []any
+		err         *mongo.CommandError
+	}{
+		"PositionIndexGreaterThanArrayLength": {
+			filter:      bson.D{{"value.5", bson.D{{"$type", "double"}}}},
+			expectedIDs: []any{},
+		},
+		"PositionIndexAtTheEndOfArray": {
+			filter:      bson.D{{"value.1", bson.D{{"$type", "double"}}}},
+			expectedIDs: []any{},
+		},
+
+		"PositionTypeDouble": {
+			filter:      bson.D{{"value.0", bson.D{{"$type", "double"}}}},
+			expectedIDs: []any{"array-double"},
+		},
+		"PositionTypeString": {
+			filter:      bson.D{{"value.0", bson.D{{"$type", "string"}}}},
+			expectedIDs: []any{"array-strings"},
+		},
+		"PositionTypeNull": {
+			filter:      bson.D{{"value.0", bson.D{{"$type", "null"}}}},
+			expectedIDs: []any{"array-null", "array-three-reverse"},
+		},
+		"PositionRegex": {
+			filter:      bson.D{{"value.0", primitive.Regex{Pattern: "foo"}}},
+			expectedIDs: []any{"array-strings"},
+		},
+		"PositionArray": {
+			filter:      bson.D{{"value.0", primitive.A{}}},
+			expectedIDs: []any{},
+		},
+
+		"NoSuchFieldPosition": {
+			filter:      bson.D{{"value.some.0", bson.A{42}}},
```

Comment:
Do we also need a test when the filter is an array and we are able to find something? (for example, if a doc stores `[42]` array in `value.some.0`?